packages/react-native-reanimated/src/jestUtils.ts

```diff
@@ -27,22 +27,81 @@ const defaultFramerateConfig = {
   fps: 60,
 };
 
+const isEmpty = (obj: object) => Object.keys(obj).length === 0;
+const getStylesFromObject = (obj: object) => {
+  return obj === undefined
+    ? {}
+    : Object.fromEntries(
+        Object.entries(obj).map(([property, value]) => [
+          property,
+          value.value ?? value,
+        ])
+      );
+};
+
+type StyleValue = { value: unknown };
+type JestInlineStylesType =
+  | {
+      [s: string]: StyleValue;
+    }
+  | ArrayLike<StyleValue>;
+
 const getCurrentStyle = (component: TestComponent): DefaultStyle => {
   const styleObject = component.props.style;
+
   let currentStyle = {};
+
   if (Array.isArray(styleObject)) {
     styleObject.forEach((style) => {
       currentStyle = {
         ...currentStyle,
         ...style,
       };
     });
```

Comment:
What in case of nested array? i.e.

`style={[styleA, styleB, [animatedStyle], styleWithInlineStyle]}`

Reply:
Using `flattenArray` doesn't solve this problem and I'm not sure if we want to dive into recursion to solve it (at least for now)

Reply:
Let's annotate it with a relevant comment then, so we see in the future that such cases aren't handled on purpose.